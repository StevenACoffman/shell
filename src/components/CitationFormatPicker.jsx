import React, { Component, PropTypes } from "react";

export default class CitationFormatPicker extends Component {
    render() {
        const { id, value, onChange, options } = this.props;

        return (
            <span>
                <label htmlFor={id}>Citation Format:</label>
                <select id={id} onChange={e => onChange(e.target.value)}
                value={value}>
                    {options.map(option =>
                        <option value={option} key={option}>
                            {option.toUpperCase()}
                        </option>)
          }
                </select>
            </span>
        );
    }
}

CitationFormatPicker.propTypes = {
    id: PropTypes.string.isRequired,
    options: PropTypes.arrayOf(
    PropTypes.string.isRequired
  ).isRequired,
    value: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired
};
