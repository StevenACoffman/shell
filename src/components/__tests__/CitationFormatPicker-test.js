import React from "react";
import CitationFormatPicker from "../CitationFormatPicker.jsx";
import renderer from "react-test-renderer";

it("renders correctly", () => {
    const tree = renderer.create(
        <CitationFormatPicker id="citation-format-picker"
                value={""}
                onChange={() => false}
                options={[]}/>
  ).toJSON();
    expect(tree).toMatchSnapshot();
});
