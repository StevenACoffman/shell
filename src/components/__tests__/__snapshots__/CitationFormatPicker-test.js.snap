exports[`test renders correctly 1`] = `
<span>
  <label
    htmlFor="citation-format-picker">
    Citation Format:
  </label>
  <select
    id="citation-format-picker"
    onChange={[Function]}
    value="" />
</span>
`;
